sale_automatic_workflow/models/sale_order.py

```diff
@@ -21,6 +21,16 @@ class SaleOrder(models.Model):
         store=True,
     )
 
+    @api.model
+    def default_get(self, fields):
+        res = super(SaleOrder, self).default_get(fields)
+        default_workflow = self.env["sale.workflow.process"].search(
+            [("default", "=", True)], limit=1
+        )
+        if default_workflow:
+            res["workflow_process_id"] = default_workflow.id
```

Comment:
You need to check if `workflow_process_id` is in the fields list to get a default value. Otherwise, this would overwrite a specific value that could have been set explicitely in the values dict passed to create function.
```suggestion
        if "workflow_process_id" in fields:
            default_workflow = self.env["sale.workflow.process"].search(
                [("default", "=", True)], limit=1
            )
            if default_workflow:
                res["workflow_process_id"] = default_workflow.id
```